src/backend/api/meals.js

```diff
@@ -1,15 +1,91 @@
 const express = require("express");
 const router = express.Router();
 const knex = require("../database");
-
-router.get("/", async (request, response) => {
+const requiredColumns = [
+  "title",
+  "description",
+  "location",
+  "when",
+  "max_reservations",
+  "created_date",
+  "price",
+];
+const valid = (input, array) => {
+  return array.every((c) => input.hasOwnProperty(c) && input[c] !== "");
+};
+router.get("/", async (req, res) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
-  } catch (error) {
-    throw error;
+    const meals = await knex("meal").select("title");
+    res.status(200).json(meals);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+});
+router.post("/", async (req, res) => {
+  const addMeal = req.body;
+  if (!valid(addMeal, requiredColumns)) {
+    return res.status(404).json("Missing required columns in the request body");
+  } else {
+    try {
+      await knex("meal").insert(addMeal);
+      res.status(201).json(`meal ${addMeal.title} added successfully`);
+    } catch (err) {
+      res.status(500).json(err);
+    }
+  }
+});
+router.get("/:id", async (req, res) => {
+  const id = req.params.id;
+  if (isNaN(id)) {
+    res.status(404).json(` ${id} not valide please provide a number`);
+  } else {
+    try {
+      const meal = await knex("meal").select("title").where({ id });
+      meal.length === 0
+        ? res.status(200).json("no meal with this id")
+        : res.status(200).json(meal);
+    } catch (err) {
+      res.status(500).json(err);
+    }
+  }
+});
+
+router.put("/:id", async (req, res) => {
+  const id = req.params.id;
+  const updatedMeal = req.body;
+  if (isNaN(id) || !valid(updatedMeal)) {
+    res.status(404).json(` ${id} not valide please provide a number`);
+  } else {
+    try {
+      updatedMeal.id = id;
+      const meal = await knex("meal").where({ id });
+      meal.length === 0
+        ? res.status(200).json(`meal not found`)
+        : (await knex("meal").where({ id }).update(updatedMeal)) &&
+          res
+            .status(200)
+            .json(`meal ${updatedMeal.title} updated successfully`);
```

Comment:
Well, this is so nice that you understand usage of ternary operators. But what I would suggest in this, it is more understandable and simple to use if conditions.
Your refactored code will look like this:
`if (meal.length === 0) res.status(200).json('meal not found');`
`await knex("meal").where({ id }).update(updatedMeal);`
`res.status(200).json('Meal ${updatedMeal.title} updated successfully');`

Reply:
and if you consider it for the rest of the endpoints, it will make code more readable and easier to understand